### PR TITLE
fix: some bug in new persona logic

### PR DESCRIPTION
**Pull request by Lanttcat**
## Description

Closes #MF-1687 #MF-1845 #MF-1681 #MF-1680 #MF-1683 #MF-1697 #MF-1696

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/mask/src/plugins/Avatar/SNSAdaptor/index.tsx [hunk 1]
```diff
@@ -25,6 +25,15 @@ const sns: Plugin.SNSAdaptor.Definition = {
                 RenderEntryComponent(EntryComponentProps) {
                     const [open, setOpen] = useState(false)
                     const clickHandler = () => setOpen(true)
+                    useEffect(
+                        () => () => {
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Avatar/SNSAdaptor/index.tsx]: What's this for? Register a listener on tear-down an effect? Why didn't we need to unregister?

File: packages/mask/src/plugins/Tips/SNSAdaptor/index.tsx [hunk 2]
```diff
@@ -43,6 +44,16 @@ const sns: Plugin.SNSAdaptor.Definition = {
                 RenderEntryComponent(EntryComponentProps) {
                     const [open, setOpen] = useState(false)
                     const clickHandler = () => setOpen(true)
+
+                    useEffect(() => {
+                        return () => {
+                            CrossIsolationMessages.events.requestOpenApplication.on(({ open, application }) => {
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Tips/SNSAdaptor/index.tsx]: Why didn't we need to unregister?

File: packages/mask/src/plugins/Avatar/SNSAdaptor/index.tsx (hunk 1 above)
Comment (UncleBill) [reply packages/mask/src/plugins/Avatar/SNSAdaptor/index.tsx]: @Lanttcat  `CrossIsolationMessages.events.requestOpenApplication.on`  will return an unsubscribe function.

Just return the expression

```ts
useEffect(() => {
    return CrossIsolationMessages.events.requestOpenApplication.on(fn)
}, [])
```

File: packages/plugins/Web3Profile/src/SNSAdaptor/index.tsx [hunk 3]
```diff
@@ -26,6 +27,15 @@ const sns: Plugin.SNSAdaptor.Definition = {
             }
             return {
                 RenderEntryComponent(EntryComponentProps) {
+                    useEffect(
```
Comment (UncleBill) [inline packages/plugins/Web3Profile/src/SNSAdaptor/index.tsx]: ditto

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**